self_unenrollment_tool/views.py

```diff
@@ -0,0 +1,245 @@
+
+from logging import getLogger
+
+from canvas_sdk import RequestContext
+from canvas_sdk.methods.enrollments import (conclude_enrollment,
+                                            list_enrollments_courses)
+from django.conf import settings
+from django.contrib import messages
+from django.http import HttpRequest, JsonResponse
+from django.shortcuts import redirect, render
+from django.urls import reverse
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_POST
+from icommons_common.models import CourseEnrollee, CourseGuest, CourseStaff
+from pylti1p3.contrib.django import (DjangoCacheDataStorage, DjangoDbToolConf,
+                                     DjangoMessageLaunch, DjangoOIDCLogin)
+
+from self_enrollment_tool.models import SelfEnrollmentCourse
+
+from .lti1p3_utils import (get_launch_url, get_message_launch,
+                           require_lti_launch)
+
+logger = getLogger(__name__)
+
+SDK_SETTINGS = settings.CANVAS_SDK_SETTINGS
+# make sure the session_inactivity_expiration_time_secs key isn't in the settings dict
+SDK_SETTINGS.pop('session_inactivity_expiration_time_secs', None)
+SDK_CONTEXT = RequestContext(**SDK_SETTINGS)
+
+
+@csrf_exempt
+def login(request: HttpRequest):
+    tool_conf = DjangoDbToolConf()
+    launch_data_storage = DjangoCacheDataStorage()
+
+    oidc_login = DjangoOIDCLogin(request, tool_conf, launch_data_storage=launch_data_storage)
+    target_link_uri = get_launch_url(request)
+    return oidc_login.enable_check_cookies().redirect(target_link_uri)
+
+
+@require_POST
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Disabling cross-site request forgery (CSRF) protection can make your users vulnerable to attacks. Enable CSRF if your application can be accessed through a browser and another token mechanism isn&#39;t in place.

[Learn more](https://cwe.mitre.org/data/definitions/352.html)

Similar issue at line numbers 43, 44, 46, 47, and 48.

Reply:
CSRF not needed here